FirebaseFunctionsSwift/Sources/Codable/Callable+Codable.swift

```diff
@@ -73,7 +73,7 @@ public struct Callable<Request: Encodable, Response: Decodable> {
   ///
   /// - Parameter data: Parameters to pass to the trigger.
   /// - Parameter completion: The block to call when the HTTPS request has completed.
-  public func call(_ data: Request,
+  public func call(_ data: Request? = nil,
```

Comment:
Hmm, how common do you think it is for people to have functions with no parameters? I worry that we're over optimizing here and the call site is less clear that no arguments are being passed (in the situation where you want that)

Reply:
I second that. If you have a truly optional parameter you just use `Param?.self` as your generic parameter. 
The only thing I can see missing is if you either never have any parameters or never have return values. In those situations it would be nice to use `Void.self` as generic params, but unfortunately `Void` does not conform to `Codable` afair.
But I also think that it's probably overkill to model an overload with void request param or void response param and one with both. Instead we could start a discussion on the swift evolution forums to argue that `Void` ought to be `Codable`.

Reply:
Reverted parameterless